### ISSUE TITLE
Extend formatting to include dbt config block at top of model file

### ISSUE THREAD

**Issue by andrewpblee** [sqlfluff/sqlfluff#2303]
From thread on SQLFluff slack: https://sqlfluff.slack.com/archives/C01GX4L213J/p1641991547059300

When using SQLFluff with dbt, it appears that sqlfluff ignores or does not lint the config block, a jinja tag at the top of dbt models that include metadata regarding the model (schema name, materialisation type etc.) 

Given that this is in the same file as the sql for a model, it would be very useful for SQLfluff to also lint this tag. Currently if I run the following query, sqlfluff returns no errors:

```
{{config(
        materialized='table',
        schema='example',
        name=
'example_dbt_prod_models')}}

SELECT CURRENT_TIMESTAMP() AS time_of_run
```

SQLFluff does already format jinja tags within the sql, in rule `L046`, so we do have precedent.

**Comment by jpy-git** [comment]
@tunetheweb FYI this is going to be very difficult for us to format atm, this is the config block that renders to an empty string and doesn't get included in the parse tree.

(do like the idea though)

**Comment by tunetheweb** [comment]
Yeah but could look at the raw, pre-templated, code. Like `L046` does.

**Comment by andrewpblee** [comment]
IMO I would want to formatter to focus on the whitespace and the newlines, and I would format like so:
from this 
```
{{config(
        materialized='table',
        schema='example',
        name=
'example_dbt_prod_models')}}
```
to this: 
```
{{
    config(
        materialized='table',
        schema='example',
        name='example_dbt_prod_models'
    )
}}
```

**Comment by FrankTub** [comment]
Also would really like this to be a seperate rule.
